context.go

```diff
@@ -0,0 +1,37 @@
+package zlog
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/baggage"
+)
+
+// ContextWithValues is a helper for the go.opentelemetry.io/otel/baggage v1
+// API. It takes pairs of strings and adds them to the Context via the baggage
+// package.
+//
+// Any trailing value is silently dropped.
+func ContextWithValues(ctx context.Context, pairs ...string) context.Context {
+	var err error
+	b := baggage.FromContext(ctx)
+	pairs = pairs[:len(pairs)-len(pairs)%2]
+	ms := append(b.Members(), make([]baggage.Member, len(pairs)/2)...)
+	for mi, pi := b.Len(), 0; mi < len(ms); mi, pi = mi+1, pi+2 {
+		k, v := pairs[pi], pairs[pi+1]
+		ms[mi], err = baggage.NewMember(k, v)
+		if err != nil {
+			Warn(ctx).
+				Err(err).
+				Str("key", k).
+				Msg("failed to create baggage member")
+			continue
+		}
+	}
+	b, err = baggage.New(ms...)
+	if err != nil {
+		Warn(ctx).
+			Err(err).
+			Msg("failed to create baggage")
```

Comment:
likewise, return `ctx` here to avoid injecting a `nil` baggage into the context (which may cause panics downstream in the worst case, and in the best case would still hide the previously existing baggage)